lib/quarantine/rspec_adapter.rb

```diff
@@ -47,21 +48,27 @@ def self.bind_record_tests
       config.after(:each) do |example|
         metadata = example.metadata
 
-        status = :passing
-        if RSpec.configuration.skip_quarantined_tests && Quarantine::RSpecAdapter.quarantine.test_quarantined?(example)
-          Quarantine::RSpecAdapter.quarantine.pass_flaky_test(example)
-          status = :quarantined
-        elsif metadata[:retry_attempts] + 1 == metadata[:retry] && example.exception
-          # will record the failed test if it's final retry from the rspec-retry gem
-          status = :failing
-        elsif (metadata[:retry_attempts] > 0 && example.exception.nil?) || metadata[:flaky]
-          # will record the flaky test if it failed the first run but passed a subsequent run;
-          # optionally, the upstream RSpec configuration could define an after hook that marks an example as flaky in
-          # the example's metadata
-          status = :quarantined
+        # optionally, the upstream RSpec configuration could define an after hook that marks an example as flaky in
+        # the example's metadata
+        quarantined = Quarantine::RSpecAdapter.quarantine.test_quarantined?(example) || metadata[:flaky]
+        if example.exception
+          if metadata[:retry_attempts] + 1 == metadata[:retry]
+            # will record the failed test if it's final retry from the rspec-retry gem
+            if RSpec.configuration.skip_quarantined_tests && quarantined
+              example.clear_exception!
+              Quarantine::RSpecAdapter.quarantine.record_test(example, :quarantined, false)
+            else
+              Quarantine::RSpecAdapter.quarantine.record_test(example, :failing, false)
```

Comment:
might be a nit: now that it has 3 args I find this method signature a bit confusing. it seems like the second argument and third argument is stating similar things. but i guess in reality the second arg is `the (previous) status of a test example we had in db` and the third arg is `did this test example pass, in this current run` ? is this understanding right?

Reply:
Sort of - the second arg is the current status, which can depend _both_ on the previous status _and_ on whether the test just passed. For example, the test can either pass or fail, but if it's currently quarantined, it'll stay quarantined (until it record 100 consecutive passes).

I've converted `passed` to a keyword arg to hopefully make things a little clearer.

Reply:
ok. thanks!